### PR TITLE
Trident v21.04 templates added 

### PR DESCRIPTION
**Pull request by ntap-troy**
I removed the 21.01 directories, added 21.04 directories, and updated all of the config files in:
- config-templates/netapp/netapp-ontap-nas/21.04
- config-templates/netapp/netapp-ontap-san/21.04
- config-templates/netapp/netapp-trident/21.04

### CODE REVIEW

File: config-templates/netapp/netapp-ontap-nas/21.04/README.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+# NetApp Trident - ONTAP NAS Driver
+
+You can use the `netapp-ontap-nas` Satellite storage template to deploy NetApp storage drives which you can use dynamically provision and mange file volumes in your ONTAP NAS.
```
Comment (derekpoindexter) [inline config-templates/netapp/netapp-ontap-nas/21.04/README.md:3]:
```suggestion
You can use the `netapp-ontap-nas` Satellite storage template to deploy NetApp storage drivers which you can use dynamically provision and manage file volumes in your ONTAP NAS.
```

File: config-templates/netapp/netapp-ontap-nas/21.04/README.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+| Storage class name | Type | File system | IOPs | Reclaim policy |
+| --- | --- | --- | --- | --- |
+| `ntap-file-gold` | Ontap-NAS | NFS | user defined**\*** | Delete |
+| `ntap-file-silver` | Ontap-NAS | NFS | user defined**\*** | Delete |
+| `ntap-file-bronze` | Ontap-NAS | NFS | user defined**\*** | Delete | 
+| `ntap-file-default` | Ontap-NAS | NFS | n/a | Delete | 
+
+**\*NOTE**: In order to use the **ntap-file-gold**, **ntap-file-silver** or **ntap-file-bronze** storage classes, the storage administrator must create the associated QoS policies on the storage controller using the following QoS Policy Group names: **gold**, **silver**, **bronze**.
+For information on creating and managing QoS Policy groups, please refer to the [ONTAP 9 Storage Management documentation](https://docs.netapp.com/ontap-9/index.jsp).
```
Comment (derekpoindexter) [inline config-templates/netapp/netapp-ontap-nas/21.04/README.md:55]:
```suggestion
For information on creating and managing QoS Policy groups, please refer to the [ONTAP 9 Storage Management documentation](https://docs.netapp.com/ontap-9/index.jsp).


```

File: config-templates/netapp/netapp-ontap-san/21.04/README.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
+# NetApp Trident - ONTAP SAN Driver
+
+You can use the `netapp-ontap-san` Satellite storage template to deploy NetApp storage drives which you can use dynamically provision and manage LUNs in your ONTAP SAN.
```
Comment (derekpoindexter) [inline config-templates/netapp/netapp-ontap-san/21.04/README.md:3]:
```suggestion
You can use the `netapp-ontap-san` Satellite storage template to deploy NetApp storage drivers which you can use dynamically provision and manage LUNs in your ONTAP SAN.
```

File: config-templates/netapp/netapp-trident/21.04/README.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
+# Netapp Trident
+
+Trident deploys in Kubernetes clusters as pods and provides dynamic storage orchestration services for your Kubernetes workloads. It has been designed from the ground up to help you meet the persistence demands of your containerized applications by using industry standard interfaces like the Container Storage Interface (CSI).
```
Comment (derekpoindexter) [inline config-templates/netapp/netapp-trident/21.04/README.md:3]:
```suggestion
Trident provides dynamic storage orchestration services for your Kubernetes workloads. It has been designed from the ground up to help you meet the persistence demands of your containerized applications by using industry standard interfaces like the Container Storage Interface (CSI).
```

File: config-templates/netapp/netapp-ontap-nas/21.04/README.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+* Create a cluster that meets the requirements for ONTAP NAS. For more information, see the [NetApp documentation](https://netapp-trident.readthedocs.io/en/stable-v21.04/support/requirements.html). Verify that your backend ONTAP cluster is configured as a Trident backend.
+   * You must have a dedicated SVM for Trident. Volumes that are created by Trident are created in this SVM.
+   * You must have one or more aggregates assigned to the SVM.
+   * You must have one or more dataLIFs for the SVM. Depending on the protocol used (NFS/iSCSI), at least one dataLIF is required.
+   * You must have NFS services enabled on the SVM.
+   * You must set up a snapshot policy on the SVM.
+* Share following details with Location Admin
+
+
+**Planning consideration for Location Admin**
```
Comment (derekpoindexter) [inline config-templates/netapp/netapp-ontap-nas/21.04/README.md:17]:
```suggestion
**Planning considerations for the Satellite Location Admin**
```

File: config-templates/netapp/netapp-ontap-nas/21.04/metadata.json [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  "tags": [
+    {
+      "name": "template-dependency",
+      "value": "netapp-trident"
+    }, {
+      "name": "template-dependency-version",
+      "value": "21.04"
+    }, {
+      "name": "template-dependency-description",
+      "value": "Install netapp-trident before installing ontap-nas driver."
```
Comment (derekpoindexter) [inline config-templates/netapp/netapp-ontap-nas/21.04/metadata.json:14]:
```suggestion
      "value": "Install the netapp-trident driver before installing the ontap-nas driver."
```

File: config-templates/netapp/netapp-ontap-san/21.04/README.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+   * You must have a dedicated SVM for Trident. Volumes that are created by Trident are created in this SVM.
+   * You must have one or more aggregates assigned to the SVM.
+   * You must have one or more dataLIFs for the SVM. Depending on the protocol used (NFS/iSCSI), at least one dataLIF is required.
+   * You must have iSCSI services enabled on the SVM.
+   * You must set up a snapshot policy on the SVM.
+* Share following details with Location Admin
+
+
+**Planning consideration for Location Admin**
+* Make sure `netapp-trident` is deployed on the target cluster
```
Comment (derekpoindexter) [inline config-templates/netapp/netapp-ontap-san/21.04/README.md:18]:
```suggestion
* Make sure `netapp-trident` is deployed on the target cluster.
```

File: config-templates/netapp/netapp-ontap-san/21.04/README.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+## Prerequisites
+
+**Planning consideration for Infra Admin**
+* Create a cluster that meets the requirements for ONTAP SAN. For more information, see the [NetApp documentation](https://netapp-trident.readthedocs.io/en/stable-v21.04/support/requirements.html). Verify that your backend ONTAP cluster is configured as a Trident backend.
+   * You must have a dedicated SVM for Trident. Volumes that are created by Trident are created in this SVM.
+   * You must have one or more aggregates assigned to the SVM.
+   * You must have one or more dataLIFs for the SVM. Depending on the protocol used (NFS/iSCSI), at least one dataLIF is required.
+   * You must have iSCSI services enabled on the SVM.
+   * You must set up a snapshot policy on the SVM.
+* Share following details with Location Admin
```
Comment (derekpoindexter) [inline config-templates/netapp/netapp-ontap-san/21.04/README.md:14]:
```suggestion
* Share following details with Location Admin.
```

File: config-templates/netapp/netapp-ontap-san/21.04/README.md [hunk 9]
```diff
@@ -0,0 +1,113 @@
+# NetApp Trident - ONTAP SAN Driver
+
+You can use the `netapp-ontap-san` Satellite storage template to deploy NetApp storage drives which you can use dynamically provision and manage LUNs in your ONTAP SAN.
+
+## Prerequisites
+
+**Planning consideration for Infra Admin**
```
Comment (derekpoindexter) [inline config-templates/netapp/netapp-ontap-san/21.04/README.md:7]:
```suggestion
**Planning considerations for the Infrastructure Admin**
```

**Review by derekpoindexter — CHANGES_REQUESTED** [review]
I left a few commit suggestions

File: config-templates/netapp/netapp-ontap-san/21.04/README.md [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+## Default storage classes
+
+The following storage classes are installed when you assign your `netapp-ontap-san` configuration to your clusters to allow you to take advantage of ONTAP's QoS features:
+
+| Storage class name | Type | File system | IOPs | Reclaim policy |
+| --- | --- | --- | --- | --- |
+| `ntap-block-gold` | Ontap-SAN | Block | user defined**\*** | Delete |
+| `ntap-block-silver` | Ontap-SAN | Block | user defined**\*** | Delete |
+| `ntap-block-bronze` | Ontap-SAN | Block | user defined**\*** | Delete | 
+| `ntap-block-default` | Ontap-SAN | Block | n/a | Delete |
```
Comment (nkkashyap) [inline config-templates/netapp/netapp-ontap-san/21.04/README.md:50]: Update the storage class names to `sat-netapp-block-gold`,`sat-netapp-block-silver`, `sat-netapp-block-bronze` and `sat-netapp-block-default`

File: config-templates/netapp/netapp-ontap-nas/21.04/README.md [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+
+The following storage classes are installed when you assign your `netapp-ontap-nas` configuration to your clusters to allow you to take advantage of ONTAP's QoS features:
+
+| Storage class name | Type | File system | IOPs | Reclaim policy |
+| --- | --- | --- | --- | --- |
+| `ntap-file-gold` | Ontap-NAS | NFS | user defined**\*** | Delete |
+| `ntap-file-silver` | Ontap-NAS | NFS | user defined**\*** | Delete |
+| `ntap-file-bronze` | Ontap-NAS | NFS | user defined**\*** | Delete | 
+| `ntap-file-default` | Ontap-NAS | NFS | n/a | Delete | 
+
```
Comment (nkkashyap) [inline config-templates/netapp/netapp-ontap-nas/21.04/README.md:53]: Update the storage class names to `sat-netapp-file-gold`,`sat-netapp-file-silver`, `sat-netapp-file-bronze` and `sat-netapp-file-default`

**Review by nkkashyap — CHANGES_REQUESTED** [review]

**Review by ntap-troy — COMMENTED** [review]
Updated README files with links to NetApp Installation & Configuration documentation

File: config-templates/netapp/netapp-trident/21.04/README.md [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+
+---
+     
+## Creating the Netapp Trident storage configuration
+
+
+**Example `sat storage config create` command**
+
+```sh
+ibmcloud sat storage config create --name "trident-conf" --template-name "netapp-trident" --template-version "21.04"
```
Comment (nkkashyap) [inline config-templates/netapp/netapp-trident/21.04/README.md:40]:
```suggestion
ibmcloud sat storage config create --name "trident-conf" --location <location id> --template-name "netapp-trident" --template-version "21.04"
```

**Review by nkkashyap — APPROVED** [review]
Looks good 
Pls update the `sat storage config create` cmd in README to pass location id

File: config-templates/netapp/netapp-ontap-san/21.04/storage-class-parameters.json [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+	  "description": "The selector label.",
+	  "displayname": "Selector",
+	  "name": "selector",
+	  "templateKeyName": [
+		  "selector"
+	  ],
+	  "required": "true",
+	  "category": "config",
+	  "type": "text"
+	}
```
Comment (nkkashyap) [inline config-templates/netapp/netapp-ontap-san/21.04/storage-class-parameters.json:32]: Define parameter for `fsType`
```
	}, {
	  "description": "The file system type.",
	  "displayname": "File System",
	  "name": "fsType",
	  "templateKeyName": [
		  "fsType"
	  ],
	  "default": "ext4",
	  "required": "false",
	  "category": "config",
	  "type": "text"
	}
```

**Review by nkkashyap — CHANGES_REQUESTED** [review]

**Review by nkkashyap — APPROVED** [review]
Looks good

**Review by derekpoindexter — APPROVED** [review]

**Pull request merged**